src/App.tsx

```diff
@@ -1,61 +1,137 @@
 import './App.scss';
+import React, { ChangeEvent, FormEvent, useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [visibleTodos, setVisibleTodos] = useState(todos);
+  const [title, setTitle] = useState('');
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [titleError, setTitleError] = useState(false);
+  const [userError, setUserError] = useState(false);
+
+  const handleChange = (
+    event: ChangeEvent<HTMLSelectElement | HTMLInputElement>,
+  ) => {
+    switch (event.target.id) {
+      case 'titleInput':
+        setTitleError(false);
+        setTitle(event.target.value);
+        break;
+
+      case 'userSelect':
+        setUserError(false);
+        setSelectedUser(getUser(+(event.target.value)));
+        break;
+
+      default:
+        break;
+    }
+  };
+
+  const handleSubmit = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!title) {
+      setTitleError(true);
+    }
+
+    if (!selectedUser) {
+      setUserError(true);
+    }
+
+    if (!title || !selectedUser) {
+      return;
+    }
+
+    const newTodo: Todo = {
+      id: Math.max(...todos.map(todo => todo.id)) + 1,
+      title,
+      userId: selectedUser ? selectedUser.id : -1,
```

Comment:
you already handle if you don't have a user `!selectedUser`
```suggestion
      userId: selectedUser.id,
```